pkg/network/event_common.go

```diff
@@ -18,6 +18,17 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/process/util"
 )
 
+type ProtocolType uint16
+
+const (
+	ProtocolUnclassified ProtocolType = iota
+	ProtocolUnknown
+	ProtocolHTTP
+	ProtocolHTTP2
+	ProtocolTLS
```

Comment:
Do we want to differentiate between TLS versions here? Or is there a sub-classification of TLS version?

Reply:
Also, should those be bitmasks instead of enum/ints?
Since we can have HTTP or HTTP2 inside of TLS?

Reply:
we talked about adding tags to the protocol in the future (after examining the performance effect)
and then we will be able to diffrentiate between the different TLS versions (or HTTP versions 0.9 1.0 1.1 ...)

we don't use bitmask because we anticipate we will have potentially many protocols (for example having AWS internal protocols SQS, SNS, etc, google's internal protocols, many DBs and we don't want to limit ourself with number of bits)

Reply:
We should discuss with backend; however, as much as possible IMO we should try to lay out the payload with an idea of what we think we'll send (i.e. HTTPS with version and included protocols), otherwise the backend has to have lots of 
```if (/* ver1 */) 
   else if (/* ver2 */)```
   etc.

Reply:
already discussed about https://dd.slack.com/archives/CFB4213BJ/p1665550049935749

Reply:
the basic idea is to have the protocol and enrichment via tags